lib/AppInfo/Application.php

```diff
@@ -22,6 +30,21 @@ public function __construct() {
 	public function register(IRegistrationContext $context): void {
 		$context->registerEventListener(BeforeUserDeletedEvent::class, UserDeletedListener::class);
 		$context->registerEventListener(DatasourceEvent::class, AnalyticsDatasourceListener::class);
+		$context->registerEventListener(RenderReferenceEvent::class, TablesReferenceListener::class);
+
+		$context->registerSearchProvider(SearchTablesProvider::class);
+
+		$container = $this->getContainer();
+		/** @var IConfig $config */
+		try {
+			$config = $container->get(IConfig::class);
+			if (version_compare($config->getSystemValueString('version', '0.0.0'), '26.0.0', '<')) {
+				$context->registerReferenceProvider(TableReferenceProvider::class);
+			} else {
+				$context->registerReferenceProvider(SearchableTableReferenceProvider::class);
+			}
+		} catch (NotFoundExceptionInterface|ContainerExceptionInterface $e) {
```

Comment:
If IConfig is not found by the container, Nextcloud is not working at all :grin: . But it does not hurt to catch the exception.

Reply:
I'm taking about the try/catch :grin: